backend/app/worker/photogrammetry/images_to_point_cloud.py

```diff
@@ -16,6 +19,129 @@ def remove_if_exists(path):
+
+def get_max_image_resolution(images_dir: str):
+    """Get the maximum resolution of images in the directory"""
+    max_width, max_height = 0, 0
+    try:
+        for filename in os.listdir(images_dir):
+            if filename.lower().endswith(('.jpg', '.jpeg', '.png', '.tif', '.tiff')):
+                img_path = os.path.join(images_dir, filename)
+                img = cv2.imread(img_path)
+                if img is not None:
+                    height, width = img.shape[:2]
+                    max_width = max(max_width, width)
+                    max_height = max(max_height, height)
+    except Exception as e:
+        logger.warning(f"Error reading image dimensions: {e}")
+        return (1920, 1080)
+    
+    if max_width == 0 or max_height == 0:
+        return (1920, 1080)
+    
+    return (max_width, max_height)
+
+def calculate_resource_allocation(images_dir: str):
+    """Calculate optimal resource allocation based on system capabilities and image resolution"""
+    available_memory_mb = memory_available()
+    cpu_count = get_cpu_count()
+
+    max_width, max_height = get_max_image_resolution(images_dir)
+    max_mp = (max_width * max_height) / 1_000_000  # Megapixels
+    
+    logger.info(f"System has {cpu_count} CPU cores and {available_memory_mb}MB available memory")
+    logger.info(f"Maximum image resolution: {max_width}x{max_height} ({max_mp:.2f}MP)")
+    
+
+    #TODO: Adjust these values based on benchmarking  specific workload
+    # memory_per_process_mb = 2000 + (20 * max_mp) 
+    # print(f"Memory per process: {memory_per_process_mb}MB")
+    # memory_limited_processes = max(1, int(available_memory_mb / memory_per_process_mb))
+    # logger.info(f"Memory-limited processes: {memory_limited_processes}")
+    
+    processes = max(1, cpu_count)
+    logger.info(f"Calculated processes: {processes}")
+    depthmap_resolution = 2048  
+    feature_process_size = 2048
+    if processes > 16 and available_memory_mb > 16000:
+        depthmap_resolution = max(4096, max(max_width, max_height))
+        feature_process_size = 4096
+    elif processes > 8 and available_memory_mb > 8000:
+        depthmap_resolution = max(2048, max(max_width, max_height)/2)
+        feature_process_size = 2048
+    else:
+        depthmap_resolution = max(1024, max(max_width, max_height)/4)
+
+    
+    depthmap_processes = max(1, int(processes / 2))  
+
+
+    return {
+        "processes": processes,
+        "read_processes": processes,
+        "depthmap_resolution": depthmap_resolution,
+        "depthmap_processes": depthmap_processes,
+        "feature_process_size": feature_process_size,
+        "memory_mb": available_memory_mb
+    }
+
+def update_config_for_stage(process_dir: str, config_updates):
+    """Update the config.yaml file for a specific pipeline stage"""
+    config_path = os.path.join(process_dir, 'config.yaml')
+    
+    with open(config_path, 'r') as f:
+        config_lines = f.readlines()
```

Comment:
Please uniform the logic to remove and create a new one config ymal ans use it also for the initial setup